tests/test_number2pencil_2.py

```diff
@@ -0,0 +1,112 @@
+            "Sheet Pan Strawberry Shortcake Recipe", self.harvester_class.title()
+        )
+
+    def test_category(self):
+        self.assertEqual("Dessert", self.harvester_class.category())
+
+    def test_total_time(self):
+        self.assertEqual(45, self.harvester_class.total_time())
+
+    def test_yields(self):
+        self.assertEqual("12 servings", self.harvester_class.yields())
+
+    def test_image(self):
+        self.assertEqual(
+            "https://www.number-2-pencil.com/wp-content/uploads/2023/05/Sheet-Pan-Strawberry-Shortcake-7.jpg",
+            self.harvester_class.image(),
+        )
+
+    def test_ingredients(self):
+        self.assertEqual(
+            [
+                "3 1/2 cups all-purpose flour",
+                "3/4 cup sugar (granulated)",
+                "5 teaspoons baking powder",
+                "1/2 teaspoon table salt",
+                "1 cup unsalted butter (cold, 2 sticks)",
+                "2 eggs (lightly beaten)",
+                "2 cups heavy whipping cream",
+                "1 teaspoon pure vanilla extract",
+                "2 cups heavy whipping cream",
+                "1/3 cup powdered sugar",
+                "2 pounds strawberries (washed and sliced)",
+                "1/4 cup sugar (granulated, more or less depending on berry sweetness)",
+                "1 teaspoon fresh lemon juice",
+                "chamomile flowers (optional)",
+            ],
+            self.harvester_class.ingredients(),
+        )
+
+    def test_ingredient_groups(self):
+        self.assertEqual(
+            [
+                IngredientGroup(
+                    ingredients=[
+                        "3 1/2 cups all-purpose flour",
+                        "3/4 cup sugar (granulated)",
+                        "5 teaspoons baking powder",
+                        "1/2 teaspoon table salt",
+                        "1 cup unsalted butter (cold, 2 sticks)",
+                        "2 eggs (lightly beaten)",
+                        "2 cups heavy whipping cream",
+                        "1 teaspoon pure vanilla extract",
+                    ],
+                    purpose="Shortcake",
+                ),
+                IngredientGroup(
+                    ingredients=[
+                        "2 cups heavy whipping cream",
+                        "1/3 cup powdered sugar",
+                    ],
+                    purpose="Whipped Cream",
+                ),
+                IngredientGroup(
+                    ingredients=[
+                        "2 pounds strawberries (washed and sliced)",
+                        "1/4 cup sugar (granulated, more or less depending on berry sweetness)",
+                        "1 teaspoon fresh lemon juice",
+                        "chamomile flowers (optional)",
+                    ],
+                    purpose="Strawberry Topping",
+                ),
+            ],
+            self.harvester_class.ingredient_groups(),
+        )
```

Comment:
@strangetom do you have any thoughts on whether it's generally a good idea for us to recommend having test coverage on both `ingredients` and `ingredient_groups` when a site supports ingredient groups?

Reply:
@jayaddison I think I like the approach here, where the ingredients and the groups are both explicitly tested. I think doing it this way makes for a simpler developer experience, especially for new contributors, and makes the tests consistent across all scrapers.

I do recognise that the combination of `test_ingredient_groups` and `test_consistent_ingredients_lists` achieve the same as `test_ingredient`, so I could be persuaded that avoiding that duplication could be a good thing.